### ISSUE TITLE
Release Workflow broken

### ISSUE THREAD

**Issue by hohwille** [devonfw/IDEasy#448]
### Expected behavior

Release can be created automatically from the click of a button.

### Actual behavior

Release-Build workflow got stuck (infinity loop?):
https://github.com/devonfw/IDEasy/actions/runs/9808961679/job/27086046702
```
 [INFO] Remote deploy finished with success.
 ...
 [INFO] BUILD SUCCESS
 ...
 Started at 1720185953000
Run maven_config="$(cat .mvn/maven.config)"
  maven_config="$(cat .mvn/maven.config)"
  current_version="${maven_config/#*-Drevision=}"
  current_version="${current_version/ */}"
  current_segment="${current_version/*./}"
  (( next_segment=${current_segment}+1 ))
  while [ ${#next_segment} -ne 3 ]; do next_segment="0${next_segment}"; done
  next_version="${current_version%.*}.${next_segment}-SNAPSHOT"
  sed -i "s/${current_version}/${next_version}/" .mvn/maven.config
  git add -f .mvn/maven.config
  git commit -m "set next version to ${next_version}"
  # prevent GH006 error when pushing to protected branch
  git push
  git push --tags
  noDotVersion="${current_version//.}"
  gh release create "release/${current_version}" ./cli/target/*.tar.gz --title "${current_version}" --notes "# Download
  * Windows: [https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/${current_version}/ide-cli-${current_version}-windows.tar.gz](https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/$%7Bcurrent_version%7D/ide-cli-$%7Bcurrent_version%7D-windows.tar.gz)
  * Mac: [https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/${current_version}/ide-cli-${current_version}-mac.tar.gz](https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/$%7Bcurrent_version%7D/ide-cli-$%7Bcurrent_version%7D-mac.tar.gz)
  * Linux: [https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/${current_version}/ide-cli-${current_version}-linux.tar.gz](https://repo1.maven.org/maven2/com/devonfw/tools/IDEasy/ide-cli/$%7Bcurrent_version%7D/ide-cli-$%7Bcurrent_version%7D-linux.tar.gz)
  # Changes
  https://github.com/devonfw/IDEasy/blob/main/CHANGELOG.asciidoc#${noDotVersion}"
  shell: /usr/bin/bash -e {0}
  env:
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    GH_TOKEN: ***
    BUILD_USER: ***
    BUILD_USER_PASSWD: ***
```
No more life-sign from here.
I ran the bash code locally on my machine and it worked apart from the gh commands.
I suspected the `while` loop to maybe run infinitely since the invariant is not smart (if the length is already above 3 characters, it will append zeros forever).
However, this does not seem to be the reason for the problem.
As always with github actions that do not work but locally all is not reproducible, they are a black box to me and pretty much get in the way blocking us...
Automation is key but it only helps if it just works flawlessly.

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. run Release workflow manually
2. wait and see the logs

### Related/Dependent Issues

#251 
#252 

### Comments/Hints:

Like with PR #424 / #444 I am clueless what could be the reason.

### Affected version:

* OS: Linux (Github action in docker)

**Comment by hohwille** [comment]
My analysis is complete: I broke the workflow myself. I copied the right block of bash code from the workflow into a local bash script that I then modified and tested until the issue I wanted to fix was solved. However, when I copied that bash code back, I put it into the workflow at the wrong position. The pitfall is that there are two blocks of bash code that both read the current maven project version and compute the next one:
* the first one is just stripping off the `-SNAPSHOT`
* the second one is increasing the last segment and then re-adding the `-SNAPSHOT`. This one was buggy before and I fixed it but copied it over the first one instead of the second one.

What also confused me and played a trick on me for this mistake was the redundant duplication of 3 lines of code so that the entire bash code had the same length as the one in the other block and uses the same variables
https://github.com/devonfw/IDEasy/blob/128bf5c3d8b3c9dd2a1a073a185aafdf60842030/.github/workflows/release.yml#L64-L70

Bottom line: Never ever edit workflows online on github even if you think you are smart ;)
